pkg/api/vlabs/validate.go

```diff
@@ -596,6 +596,14 @@ func (a *Properties) validateExtensions() error {
 		if len(agentPool.Extensions) != 0 && (len(agentPool.AvailabilityProfile) == 0 || agentPool.IsVirtualMachineScaleSets()) {
 			return errors.Errorf("Extensions are currently not supported with VirtualMachineScaleSets. Please specify \"availabilityProfile\": \"%s\"", AvailabilitySet)
 		}
+
+		if agentPool.OSType == Windows && len(agentPool.Extensions) != 0 {
```

Comment:
thanks for the PR @vyta ! Could you please add a small unit test to test this new validation?

Reply:
https://github.com/Azure/aks-engine/blob/master/pkg/api/vlabs/validate_test.go#L865

Reply:
Added--let me know if that works